pkg/build/generator/generator.go

```diff
@@ -732,6 +732,12 @@ func generateBuildFromBuild(build *buildapi.Build, buildConfig *buildapi.BuildCo
 		// builds without a buildconfig don't have build numbers.
 		delete(newBuild.Annotations, buildapi.BuildNumberAnnotation)
 	}
+
+	// bz1371047: if they exist, Jenkins reporting annotations must be removed when cloning.
+	delete(newBuild.Annotations, buildapi.BuildJenkinsStatusJSONAnnotation)
+	delete(newBuild.Annotations, buildapi.BuildJenkinsLogURLAnnotation)
+	delete(newBuild.Annotations, buildapi.BuildJenkinsBuildURIAnnotation)
```

Comment:
seems like maybe there are some other annotations we should be clearing when cloning a build too, like the podname annotation.